common/src/main/java/com/wynntils/handlers/inventory/PendingInteraction.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.handlers.inventory;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.handlers.inventory.event.InventoryInteractionEvent;
+import it.unimi.dsi.fastutil.ints.IntList;
+import java.util.List;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+
+interface PendingInteraction {
+    boolean onSlotUpdate(int containerId, int slotNum, ItemStack newStack);
+
+    default boolean onContentUpdate(int containerId, List<ItemStack> newInv) {
+        return true;
+    }
+
+    static boolean areItemsSimilar(ItemStack a, ItemStack b) {
```

Comment:
I was about to suggest that you put this in `ItemStackUtils`, but then I double checked and to my surprise we did not have such a class. We really ought to have one, in `com.wynntils.utils.utils.mc`. I suggest you create one and move this function there.